### PR TITLE
Add Dockerfile and .dockerignore

### LINKED ISSUE #8: Create Dockerfile
An easy way of distributing and running server applications is by using docker. To package our application as an image for docker we need a Dockerfile with layer commands.

We can build an image using the files in target folder that can be created by running mvn commands or we can make a multistage Dockerfile so the build process runs in a docker container. Using this technique we don't need java or maven installed on the computer running the build process.

I therefor suggests creating a multistage Dockerfile in root folder of this project. 
When copying project files to the build image we can avoid copying files not needed for the build process. To specify which files we can create a .dockerignore file in the root folder. https://shisho.dev/blog/posts/how-to-use-dockerignore/

### PR DESCRIPTION
**Pull request by vonfredd**
---
"Add Dockerfile and .dockerignore"
---

### Related Issue(s)
Closes #8 

### Proposed Changes
- **Change 1**: [Add Dockerfile to project root]
- **Change 2**: [Add .dockerignore to project root]

### Description
Added files and syntax to be able to containerize the application while excluding unnecessary files from adding size to the Docker image.

### CODE REVIEW

**Review by ByHatice — DISMISSED** [review]

**Review by Win-ther — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,8 @@
+FROM maven:3.9.6-eclipse-temurin-21-alpine AS  build
+COPY ./ /src
+RUN mvn -f /src/pom.xml clean package
+
+FROM eclipse-temurin:21_35-jre-alpine
```
Comment (kappsegla) [inline Dockerfile:5]: Version 21_35 is an "old" version of Java version 21. Latest exact version number is 21.0.2+13 https://adoptium.net/temurin/archive/

So I suggest that we use 21.0.2_13-jre-alpine instead to have an exact tag which is prefered. If we want a tag that always tracks the latest 21 release we could instead use 21-jre-alpine.

**Review by kappsegla — CHANGES_REQUESTED** [review]
Suggest some minor changes to formatting and versions.

File: .dockerignore [hunk 2]
```diff
@@ -0,0 +1,5 @@
+target/
+.github/
+.idea/
+.git/
+.DS_Store/
```
[the remark this answers is not in the document]
Comment (vonfredd) [reply .dockerignore]: Added newline.

File: Dockerfile (hunk 1 above)
Comment (vonfredd) [reply Dockerfile]: Updated version as suggested.

**Review by kappsegla — APPROVED** [review]
Dockerfile looks good now. Might have to introduce other changes later when adding libraries or wanting to create a custom java jre using jlink to save on space.

**Review by Win-ther — APPROVED** [review]

**Pull request merged**